Applications/WebPortal/WebApp/library/source/pages/group_page/group_controller.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react';
+import * as Router from 'react-router-dom';
+import { DisplaySize, LoadingPage, PageNotFoundPage } from '../..';
+import { GroupInfoController, GroupInfoModel, LocalGroupInfoModel }
+  from './group_info_page';
+import { GroupModel } from './group_model';
+import { GroupPage } from './group_page';
+import { GroupSubPage } from './group_sub_page';
+
+interface Properties {
+
+  /** Determines the layout to use based on the display device. */
+  displaySize: DisplaySize;
+
+  /** The model representing the group to display. */
+  model: GroupModel;
+}
+
+interface State {
+  isLoaded: boolean;
+  cannotLoad: boolean;
+  redirect: string;
+  groupInfoModel: GroupInfoModel;
+}
+
+/** Implements a controller for the GroupPage. */
+export class GroupController extends React.Component<Properties, State> {
+  constructor(props: Properties) {
+    super(props);
+    this.state = {
+      isLoaded: false,
+      cannotLoad: false,
+      redirect: null,
+      groupInfoModel: new LocalGroupInfoModel([])
```

Comment:
What does this ```GroupInfoModel``` do? It looks like it's not used for anything and instead you can work with ```model.accounts``` directly.

Reply:
It's used as a prop for rendering GroupInfoController. Should I instead create a GroupInfoModel getter on HttpGroupModel and use that for loading the prop to the GroupInfoController?

Reply:
Change ```GroupInfoController``` to take in the ```model.accounts``` as a prop.

Reply:
Would I still be using GroupInfoModel in this case, or I can ignore it and the isLoaded state variable in GroupInfoController?

Reply:
No I don't see a need for a model in this case. You would ignore it entirely.